src/TodoApp.tsx

```diff
@@ -0,0 +1,162 @@
+import React, {
+  FormEvent,
+  useEffect,
+  useState,
+} from 'react';
+import { useLocation } from 'react-router-dom';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { FilterTypes } from './types/FilterTypes';
+import { Notification } from './components/Notification';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(() => {
+    const localTodos = localStorage.getItem('todos');
+
+    return localTodos ? JSON.parse(localTodos) : [];
+  });
+
+  const [title, setTitle] = useState('');
+  const [notification, setNotification] = useState<string>('');
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  const filteredTodos = todos.filter(({ completed }) => {
```

Comment:
Think about using [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) to memoize the variables